<launch>
    <arg name="dataset_name" default="s1_ccw_z_230208_o0_e0"/>
    
    <arg name="rate" default="20.0"/>
    <arg name="cruncher_node_name" default="vpr_cruncher"/>
    <arg name="plotter_node_name" default="vpr_plotter"/>
    <arg name="anon" default="True"/>

    <arg name="database_path" default="$(find aarapsi_intro_pack)/data/compressed_sets"/>
    <arg name="ref_root" default="$(find aarapsi_intro_pack)/data/$(arg dataset_name)/"/>
    <arg name="ref_imgs_path" default="[$(arg ref_root)forward,$(arg ref_root)left,$(arg ref_root)right,$(arg ref_root)panorama]"/>
    <arg name="ref_odom_path" default="$(arg ref_root)odometry.csv"/>

    <arg name="image_topic_in" default="/ros_indigosdk_occam/image0"/>
    <arg name="odometry_topic_in" default="/odometry/filtered"/>

    <arg name="img_dims" default="(64,64)"/>
    <arg name="ft_type" default="1"/>

    <arg name="tol_mode" default="2"/>
    <arg name="tol_thresh" default="1.0"/>

    <arg name="time_history_length" default="20"/>
    <arg name="icon_info" default="(50,20)"/>
    <arg name="namespace" default="/vpr_nodes"/>
    <arg name="frame_id" default="base_link"/>

    <arg name="cruncher_compress_in" default="True"/>
    <arg name="cruncher_compress_out" default="False"/>
    <arg name="cruncher_do_plotting" default="False"/>
    <arg name="cruncher_make_images" default="True"/>
    <arg name="cruncher_groundtruth" default="True"/>
    <arg name="cruncher_make_labels" default="True"/>

    <arg name="plotter_compress_in" default="False"/>
    
    <node pkg="aarapsi_intro_pack" type="vpr_all_in_one.py" name="$(arg cruncher_node_name)" output="screen" 
        args="$(arg dataset_name) $(arg database_path) $(arg ref_imgs_path) $(arg ref_odom_path)
              $(arg image_topic_in) $(arg odometry_topic_in) 
              -Ci $(arg cruncher_compress_in) -Co $(arg cruncher_compress_out) -P $(arg cruncher_do_plotting) -I $(arg cruncher_make_images) -G $(arg cruncher_groundtruth) -L $(arg cruncher_make_labels)
              -r $(arg rate) -l $(arg time_history_length) -i $(arg img_dims) -F $(arg ft_type) 
              -t $(arg tol_mode) -T $(arg tol_thresh) -p $(arg icon_info) 
              -N $(arg cruncher_node_name) -a $(arg anon) -n $(arg namespace) -f $(arg frame_id)">
    </node>
    
    <node pkg="aarapsi_intro_pack" type="vpr_bokeh_server.py" name="$(arg plotter_node_name)" output="screen" 
        args="$(arg dataset_name) $(arg database_path)
              -Ci $(arg plotter_compress_in) -r $(arg rate)
              -N $(arg plotter_node_name) -a $(arg anon) -n $(arg namespace)">
    </node>

</launch>